R/utils.R

```diff
@@ -140,3 +140,107 @@ edit.data.table = function(name, ...) {
   setDT(NextMethod('edit', name))[]
 }
 # nocov end
+
+col_helper = function(x, colsub, origin_call = NULL) {
+  ## helper that could replace some common code used for by, .SDcols, j, duplicated, unique, dcast, 
+  ## or anything that looks for columns
+  
+  #' @param x required; a data.frame, data.table, tibble, or list.
+  #' @param colsub takes the same arguments as .SDcols such as
+  #' `patterns()`, `var1:var2`, a variable, or a vector of characters.
+  #' When TRUE, will return seq_len(length(x))
+  #' @param origin_call takes character vector of length 1. In the short-term
+  #' there will be different modes based on the call. In the long-term, it's just
+  #' to help the end-user what caused the error
+  #' @return Integer vector with an attribute of whether negation was in the call
```

Comment:
great to see internal documentation like this!

I would be more explicit here "a boolean attribute, `"negate"`"